kolibri/core/content/utils/annotation.py

```diff
@@ -251,3 +293,41 @@ def set_availability(channel_id, checksums=None):
 
     set_leaf_node_availability_from_local_file_availability(channel_id)
     recurse_availability_up_tree(channel_id)
+    calculate_channel_fields(channel_id)
+
+
+def annotate_content(channel_id, checksums=None):
+    if checksums is None:
+        set_local_file_availability_from_disk()
+    else:
+        mark_local_files_as_available(checksums)
+
+    set_leaf_node_availability_from_local_file_availability(channel_id)
+    recurse_availability_up_tree(channel_id)
+    recurse_availability_coach_content(channel_id)
+    calculate_channel_fields(channel_id)
+
+
+def calculate_channel_fields(channel_id):
+    channel = ChannelMetadata.objects.get(id=channel_id)
+    calculate_published_size(channel)
+    calculate_total_resource_count(channel)
+    calculate_included_languages(channel)
+
+
+def calculate_published_size(channel):
+    content_nodes = ContentNode.objects.filter(channel_id=channel.id)
+    channel.published_size = _total_file_size(_files_for_nodes(content_nodes).filter(available=True))
+    channel.save()
+
+
+def calculate_total_resource_count(channel):
+    content_nodes = ContentNode.objects.filter(channel_id=channel.id)
+    channel.total_resource_count = content_nodes.filter(available=True).exclude(kind=content_kinds.TOPIC).count()
```

Comment:
Does studio dedupe by `content_id`? If it doesn't, I kind of think it should, and so should Kolibri.